### ISSUE TITLE
[e2e test failure][sig-network] Proxy version v1 should proxy through a service and a pod [Conformance] 

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#51128]
This test has started failing on our e2e test suite, erroring out with "Failed to get endpoints..."

https://k8s-testgrid.appspot.com/release-master-blocking#gce
https://k8s-testgrid.appspot.com/release-master-blocking#gci-gce

Example failures

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/proxy.go:276
Expected error:
    <*errors.errorString | 0xc420d7c9d0>: {
        s: "Failed to get endpoints for e2e-tests-proxy-h6zvv/proxy-service-28w55",
    }
    Failed to get endpoints for e2e-tests-proxy-h6zvv/proxy-service-28w55
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/proxy.go:155
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce/12667#sig-network-proxy-version-v1-should-proxy-through-a-service-and-a-pod-conformance
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/13134#sig-network-proxy-version-v1-should-proxy-through-a-service-and-a-pod-conformance

cc @kubernetes/release-maintainers 
cc @kubernetes/sig-network-bugs

**Comment by joelsmith** [comment]
This may be related to our recently-merged endpoints change in #50934. I'll look into it.

**Comment by joelsmith** [comment]
I verified that the flakiness introduced was a result of #50934. A fix in #51144 should be merging soon.
/assign @joelsmith
